src/pieces/pawn.cpp

```diff
@@ -0,0 +1,251 @@
+#include "pieces/Pawn.h"
+#include "Board.h"
+
+using Chess::PawnPiece;
+using Chess::Movement;
+using Chess::utilities::Color;
+using Chess::utilities::PieceType;
+using Chess::utilities::Direction;
+using Chess::utilities::DirectionOffset;
+
+PawnPiece::PawnPiece(Coordinate coordinate, Color color, PieceType type) : Piece{coordinate,color,type}
+{
+    symbol = (color == Color::black) ? 'P': 'p';
+}
+std::list<Movement> PawnPiece::get_pseudo_valid_movements(Board& board)
+{
+    std::list<Movement> pseudo_moviments;
+    Coordinate test_coordinate;
+    std::pair<Coordinate,Coordinate> possible_movements;
+    if(this->color == Color::white)
+    {
+        //if it's the pawn's first move, it can move two squares straight forward
+        test_coordinate = double_step(board, Color::white);
+        pseudo_moviments.push_back({this->coordinate,test_coordinate,false,false,false,false});
+        //otherwise it can move only one square straight forward
+        test_coordinate = one_step(board,this->coordinate,Color::white);
```

Comment:
se one_step ritornasse this->coordinate, nella lista dei pseudo_valid_movements aggiungi un movimento "inutile", sbagliato. Fai un if